db_stress_tool/db_stress_test_base.cc

```diff
@@ -2423,6 +2427,80 @@ Status StressTest::TestGetPropertiesOfAllTables() const {
   return db_->GetPropertiesOfAllTables(&props);
 }
 
+void StressTest::TestGetUpdatesSince(ThreadState* thread) {
+  Status s = db_->DisableFileDeletions();
+  if (!s.ok()) {
+    return;
+  }
+
+  const SequenceNumber since_seq =
+      thread->rand.Next64() % (db_->GetLatestSequenceNumber() + 1);
+  std::unique_ptr<TransactionLogIterator> iter;
+
+  s = db_->GetUpdatesSince(since_seq, &iter);
+
+  if (!s.ok()) {
+    fprintf(stderr,
+            "Failed to call GetUpdatesSince(): %s with specified "
+            "seq num: %" PRIu64 "\n",
+            s.ToString().c_str(), since_seq);
+    thread->shared->SafeTerminate();
+  }
+
+  if (!iter->status().ok()) {
+    fprintf(stderr,
+            "Iterator with non-ok status returned by "
+            "GetUpdatesSince(): %s \n",
+            iter->status().ToString().c_str());
+    thread->shared->SafeTerminate();
+  } else if (!iter->Valid()) {
+    return;
```

Comment:
If I am not wrong, then file deletions should be re-enabled here again by calling `db_->EnableFileDeletions();`. 
Otherwise this test would disable file deletions and not turn them back on, so it may influence the behavior of other tests.

Reply:
Great catch. I think ideally this special case would be deleted and we just let control flow through the end of the function per https://github.com/facebook/rocksdb/pull/12646/files#r1628634984. Still, thanks for demonstrating how seemingly small/innocent extra code can make things subtly wrong.

Reply:
Good catch! Early return needs to be handled carefully with clean up work…